File: app/src/main/java/fr/xgouchet/chronorg/ui/presenters/PortalListPresenter.java
package fr.xgouchet.chronorg.ui.presenters;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.deezer.android.counsel.annotations.Trace;

import fr.xgouchet.chronorg.data.models.Portal;
import fr.xgouchet.chronorg.data.models.Project;
import fr.xgouchet.chronorg.data.repositories.PortalRepository;
import rx.Observable;

/**
 * @author Xavier Gouchet
 */
@Trace
public class PortalListPresenter extends BaseListPresenter<Portal> {

    @NonNull private final PortalRepository eventRepository;
    @Nullable private Project project;

    public PortalListPresenter(@NonNull PortalRepository portalRepository) {
        this.eventRepository = portalRepository;
    }

    public void setProject(@NonNull Project project) {
        this.project = project;
    }

    @Override protected Observable<Portal> getItemsObservable() {
        return eventRepository.getPortalsInProject(project.getId());
    }
}

File: khronorg/src/main/java/fr/xgouchet/khronorg/feature/portals/PortalListPresenter.kt
package fr.xgouchet.khronorg.feature.jumps

import fr.xgouchet.khronorg.commons.query.QueryBuilder
import fr.xgouchet.khronorg.commons.repositories.BaseRepository
import fr.xgouchet.khronorg.feature.portals.Portal
import fr.xgouchet.khronorg.feature.projects.Project
import fr.xgouchet.khronorg.provider.KhronorgSchema.Companion.COL_PROJECT_ID
import fr.xgouchet.khronorg.ui.navigators.Navigator
import fr.xgouchet.khronorg.ui.presenters.BaseListPresenter
import io.reactivex.Observable

/**
 * @author Xavier F. Gouchet
 */
class PortalListPresenter(val repository: BaseRepository<Portal>, val project: Project, navigator: Navigator<Portal>)
    : BaseListPresenter<Portal>(navigator) {


    override fun getItemsObservable(): Observable<Portal> {
        val alteration = QueryBuilder.where {
            equals(COL_PROJECT_ID, project.id.toString())
        }

        return repository.getWhere(alteration)
    }

}